lib/context/scenario_state.rb

```diff
@@ -113,5 +113,13 @@ def get_current_material_name
       @scenario_store.get "current_material_name"
     end
 
+    def set_current_label(label)
+      @scenario_store.put "current_label", label
+    end
+
+    def get_current_label
+      @scenario_store.get "current_label"
+    end
```

Comment:
Seen ruby's openstruct? https://ruby-doc.org/stdlib-2.0.0/libdoc/ostruct/rdoc/OpenStruct.html

Reply:
Just did :), will be working on it in a separate PR.